### PR TITLE
add decorate class

### PR DESCRIPTION
**Pull request by EyasuTeshome**
- Created a class `Nameable` with a method `correct_name` that does:

> Checks the first letter of the given name is a capital letter.
> Checks the name has a maximum of 10 characters.
> Set an instance of corrector in person on initialization.

- In person added a method validate_name that uses the nameable instance to validate and save the corrected name.

### CODE REVIEW

File: school/decorate.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name(name)
+    name.capitalize![0..9]
```
Comment (BertrandConxy) [inline school/decorate.rb]: - [ ] From the project requirements, this method is supposed to rails `NotImplementedError`
image
Please correct it 😃 🏃

File: school/person.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require_relative 'decorate'
 class Person
```
Comment (BertrandConxy) [inline school/person.rb]: - [ ] From the project requirements, the Person class is supposed to be inherited from the Nameable class 👇 
image

In addition, it should define a `correct_name` method that returns the name attribute. Please implement it 😃 👍

HINT 💡 
```
class Person < Nameable
   ...

  def correct_name
      name
  end
end
```

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @EyasuTeshome, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️

- [ ] Good job so far 👏. However, it looks like there are some decorators that are in the project requirements that you have not implemented yet. Take a look at the screenshot below 👇 
image
Please do your best in order to implement them as they are part of the project requirements 😃 👍 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Josh,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please implement all the changes requested by the previous code reviewer. 

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by EyasuTeshome** [comment]
@SajjadAhmad14 please can you specify the change I didn't make? Because I tried to fix both comments given by the previous code reviewer. and please try to run check.rb to see the decorator in action.

**Comment by SajjadAhmad14** [comment]
@EyasuTeshome I could not see your changes and commits here on PR. Kindly push the changes you made and ask for a re-review once you are satisfied with the changes made!

**Comment by EyasuTeshome** [comment]
@SajjadAhmad14 I am sorry the problem was on my side. I pushed it but there was a connection error  I think, there was a message `fatal: unable to access 'https://github.com/EyasuTeshome/school-library.git/': Failed to connect to github.com` but I didn't pay attention to it. Sorry for that I have fixed it now.

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

## Hey @EyasuTeshome :wave:

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary
:heavy_check_mark: Followed Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**